GS on Personal A/B: Neutralise style variation wording in assembler

dsas opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/dotcom-forge/issues/2860

## Proposed Changes

We're changing the style variation related wording in the assembler to always say 'Custom' rather than 'Premium' to remove the association with the Premium plan. This will allow us to experiment with paywalling style variations behind different plans.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Use calypso live link
2. Go through the /start process on a free plan until you reach the design picker grid
3. Go to the bottom and use the assembler CTA
4. It should have a nav item that says Colours, click that
5. The page should have a heading that says 'Custom styles' with a star next to it, hovering over the Star should make that expand to say 'Upgrade'
6. Press 'Save colors', and repeat for fonts.

Repeat again with a Premium plan and this time the Upgrade star should not appear

Repeat both scenarios again but choose a non-virtual theme without any style variations and adding this GET param: `flags=signup/design-picker-preview-colors`, the results should be unsurprising.

Before | After (free) | After (premium)
-------|--------------|--------------
<img width="349" alt="Screenshot 2023-06-30 at 14 34 30" src="https://github.com/Automattic/wp-calypso/assets/93301/5ac114ff-b808-41fa-a3dd-426d8fd324c3"> | <img width="350" alt="Screenshot 2023-07-03 at 17 26 36" src="https://github.com/Automattic/wp-calypso/assets/93301/d472eae8-139a-4624-bd40-adc9c97343f7"> | <img width="350" alt="Screenshot 2023-07-03 at 17 25 54" src="https://github.com/Automattic/wp-calypso/assets/93301/5612fb61-5be9-4745-89f8-50052fad2551">
<img width="349" alt="Screenshot 2023-06-30 at 14 34 58" src="https://github.com/Automattic/wp-calypso/assets/93301/45b35333-6bad-418a-9b6d-7a58eaf81bc3"> | <img width="350" alt="Screenshot 2023-07-03 at 17 23 11" src="https://github.com/Automattic/wp-calypso/assets/93301/c1b3da1e-eb38-4775-b1d8-ea6f9d920c70"> | <img width="350" alt="Screenshot 2023-07-03 at 17 24 56" src="https://github.com/Automattic/wp-calypso/assets/93301/2c542015-145b-4d59-9b4e-18d50e1c631a"> 
<img width="553" alt="image" src="https://github.com/Automattic/wp-calypso/assets/93301/543cf234-bfa2-413b-876f-e9e0cc7b9042">| <img width="553" alt="Screenshot 2023-07-03 at 18 39 12" src="https://github.com/Automattic/wp-calypso/assets/93301/1d09b9f9-50dc-4153-af3e-9950794c1c25"> | <img width="553" alt="Screenshot 2023-07-03 at 18 34 12" src="https://github.com/Automattic/wp-calypso/assets/93301/a3cea391-5fd9-4ce5-8db6-553dd007a1d7">

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

dsas committed:
Neutralise style variation wording in theme details

We're changing the style variation wording to always use 'Custom' rather
than 'Premium', to remove the association with the Premium plan. This
will allow us to experiment with paywalling style variations behind
different plans.

Copons commented:
We were more oriented towards "Upgrade" though, but I've noticed that this label is not dependent on the current plan (meaning: it is displayed even if the site is on a Premium plan).

But "Paid" is slightly inappropriate too, as it doesn't require an additional payment on Premium plans, but at least it's not completely inaccurate as "Upgrade".

Do you recall why it was decided to show the label regardless of the plan?

dsas commented:
All good points on the badge.

> Do you recall why it was decided to show the label regardless of the plan?

No, it's been there in one way or another since lego implemented it forever ago, well March at least: pekYwv-Vw-p2. My made-up post-hoc justification might be that it helps to demonstrate the value that you're getting out of the plan, but we do that in other places, it wouldn't be very clear and it doesn't work with this wording anyway.

dsas commented:
The same premise is applied throughout the assembler UI, in the DP if you're on premium then we display one big happy list of style variations, but on the assembly we still have the separate free (default) list and premium list.

dsas commented:
> Do you recall why it was decided to show the label regardless of the plan?

It's not just the label, and the separate style lists, it's also the intro text 🤔 that remains saying "when you upgrade". I think it's likely that the "already on premium+" case was just overlooked. 

In the near term then I think we talk with Lego/Lucas about bringing assembler in-line with the style variation chooser. These inconsistencies are only going to become more visible as colours & fonts for all themes gets deployed (pbxlJb-3Ze-p2)


For a quick win to keep the A/B test moving we should:
 * Drop the star when the site has a global styles supported plan.
 * Change the star wording to say "Upgrade" when the site does not have a global styles supported plan.

mmtr commented:
> For a quick win to keep the A/B test moving we should:
>
> - Drop the star when the site has a global styles supported plan.
> - Change the star wording to say "Upgrade" when the site does not have a global styles supported plan.

Yeah, looks like this is the best way to proceed.

dsas committed:
Only display 'Upgrade' in Assembler when necessary

Change the PremiumBadge as used in colours and fonts on the Assembler to
say 'Upgrade' rather than 'Paid' and don't display it at all when an
upgrade isn't necessary.

dsas committed:
Use new API

dsas committed:
Remove overenthusastic context provider prop - that is not how it works

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8128315

Thank you @dsas for including a screenshot in the description! This is really helpful for our translators.

dsas committed:
Add missing font in dp prop

Copons reviewed (approved):
I'm not a fan of prop drilling, but I'm well aware of the no-import rule on packages, and I can see a baroque use of the Context API: both make a DRY approach so much more complicated here.

So: LGTM!

dsas commented:
Cheers. Yeah, I threw away my initial prop-drilling approach last week when I saw how deep the drill had to go. Then I sheepishly restored it today because it seems like the least most complicated option.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8128315).
